src/server/dfly_main.cc

```diff
@@ -199,6 +217,13 @@ Usage: dragonfly [FLAGS]
   CHECK_LT(kver.major, 99u);
   dfly::kernel_version = kver.kernel * 100 + kver.major;
 
+  string pidfile_path = GetFlag(FLAGS_pidfile);
```

Comment:
the feature is a bit more complicated than that. You also need to remove the file upon exit.
see https://github.com/redis/redis/blob/d00b8af89265c501fb4a0cdd546702b90432a896/src/server.c#L421

for example

Reply:
Can you elaborate on your comment here? I'm unsure where to place the file deletion in dfly_main.cc and how to ensure the deletion function will be called at the server exit.

Reply:
what you did at lines 265-267 is good. Was it always there? I did not see it before.

Reply:
It just added in the latest commit after you asked for the removal of the file upon exit.